crates/circuit/src/vf2.rs

```diff
@@ -955,17 +963,14 @@ struct State<G: GraphBase, OtherId> {
     /// not yet mapped, the other index is stored as `I::max()`.
     mapping: Vec<OtherId>,
     /// Mapping from node index to the generation at which a node was first added to the mapping
-    /// that had an outbound edge to that index.  This can be used to find new candidate nodes to
-    /// add to the mapping; you typically want your next node to be one that has edges linking it to
-    /// the existing mapping (but isn't yet _in_ the mapping).
-    out: Vec<usize>,
-    /// Same as `out`, except we're tracking the nodes that have an edge from them _into_ the
-    /// mapping.  This isn't used if the graph is undirected, since it'd just duplicate `out`.
-    ins: Vec<usize>,
-    /// The number of non-zero entries in `out`.
-    out_size: usize,
-    /// The number of non-zero entries in `in`.  This is always zero for undirected graphs.
-    ins_size: usize,
+    /// that had an `[outbound, inbound]` edge to that index.  This can be used to find new
+    /// candidate nodes to add to the mapping; you typically want your next node to be one that has
+    /// edges linking it to the existing mapping (but isn't yet _in_ the mapping).
+    neighbor_since: Vec<[Option<NonZero<usize>>; 2]>,
+    /// The number of neighbors to the mapping in the `[outgoing, incoming]` direction.
+    // TODO: currently this count isn't decremented when a neighbor is added to the mapping, so it
+    // becomes less useful as a metric as the mapping completes.
```

Comment:
Are you worried that the extra computation cost outweighs the benefit?

Reply:
The behaviour of not decrementing as the mapping is built up is historical, not new in the PR, and the todo was most likely more a note to myself to see if it was worth being smarter about it.